examples/free-identifier-case.kl

```diff
@@ -2,48 +2,31 @@
 
 (import "let.kl")
 (import (shift "prelude.kl" 1))
-(import (shift "quasiquote.kl" 1))
-
-(meta
-  (define identifier?
-    (lambda (x)
-      (syntax-case x
-        [(ident x) (true)]
-        [_ (false)]))))
+(import (shift "identifier.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
 
 (define-macros
   ([free-identifier-case
     (lambda (stx)
-      (syntax-case stx
-        [(cons _ body)
-         (syntax-case body
-           [(cons scrut cases)
-            (pure
-             (quasiquote/loc stx
-               (let (x ,scrut) (free-identifier-case-aux x ,cases))))])]))]
+      (syntax-case stx ()
+        [(_ ,scrut ,cases ...)
+         (pure (replace-loc stx
+                 `(let (x ,scrut) (free-identifier-case-aux x ,cases ...))))]))]
    [free-identifier-case-aux
     (lambda (stx)
-      (syntax-case stx
-        [(list (_ scrut cases))
-          (syntax-case cases
-            [() (pure '(syntax-error '"Nothing matched"))]
-            [(cons c cs)
-             (syntax-case c
-               [(list (test val))
-                (syntax-case test
-                  [(list (e x))
-                   (>>= (free-identifier=? e 'else)
-                        (lambda (eq)
-                          (if eq
-                              (pure (quasiquote/loc c (let (,x ,scrut) ,val)))
-                              (syntax-error test))))]
-                  [(ident id)
-                   (pure
-                    (quasiquote/loc c
-                     (>>= (free-identifier=? ,scrut ',id)
-                          (lambda (eq)
-                            (if eq
-                                ,val
-                                (free-identifier-case-aux ,scrut ,cs))))))])])])]))]))
+      (syntax-case stx (else)
+        [(_ ,scrut)
+         (pure '(syntax-error '"Nothing matched"))]
+        [(_ ,scrut ((else ,x) ,val))
```

Comment:
note how much simpler it now is to match on the `else` keyword!

Reply:
speaking of the `else` keyword, I didn't have to define it in a separate module and then import it twice because `else` is provided by `prelude.kl`.